src/java/io/bazel/rulesscala/test_discovery/FilteredRunnerBuilder.scala

```diff
@@ -0,0 +1,34 @@
+package io.bazel.rulesscala.test_discovery
+
+import java.util.regex.Pattern
+import org.junit.runner.Runner
+import org.junit.runners.BlockJUnit4ClassRunner
+import org.junit.runners.model.FrameworkMethod
+import org.junit.runners.model.RunnerBuilder
+import scala.collection.JavaConverters._
+
+object FilteredRunnerBuilder extends RunnerBuilder {
+  // Defined by --test_filter bazel flag.
+  private def pattern =
+    sys.env.get("TESTBRIDGE_TEST_ONLY").map(Pattern.compile(_))
+  override def runnerForClass(testClass: Class[_]): Runner = {
+    new BlockJUnit4ClassRunner(testClass) {
```

Comment:
I think the compiler should be smart enough to handle this, but I'll pull it out just in case.